auraed/src/lib.rs

```diff
@@ -199,24 +205,38 @@ pub async fn run(
         })?;
 
         // Install eBPF probes in the host Aurae daemon
-        let (_bpf_scope, signals) = if context == AuraeContext::Cell
+        let (_aurae_bpf, posix_signals_listener) = if context
+            == AuraeContext::Cell
             || context == AuraeContext::Container
         {
             (None, None)
         } else {
             // TODO: Add flags/options to "opt-out" of the various BPF probes
             info!("Loading eBPF probes");
-            let mut bpf_loader = BpfLoader::new();
-            let listener = bpf_loader
-                .read_and_load_tracepoint_signal_signal_generate()
-                .ok();
 
-            if listener.is_none() {
-                warn!("Missing eBPF probe. Skipping signal reporting.");
-            }
-
-            // Need to move bpf_loader out to prevent it from being dropped
-            (Some(bpf_loader), listener)
+            let mut bpf = match AuraeBpf::load() {
+                Ok(bpf) => Some(bpf),
+                Err(e) => {
+                    warn!("eBPF: Cannot load probes: {e}");
+                    None
+                }
+            };
+
+            let posix_signals_listener = if let Some(bpf) = &mut bpf {
+                let posix_signals_listener = match bpf.load_and_attach_tracepoint_program::<SignalSignalGenerateTracepointProgram, _>() {
```

Comment:
If we map each program to a gRPC service, the health endpoint can be used to check if the program is loaded by checking if the service is being served.

Reply:
This won't be the case. As we speak, I am adding at least 2 but likely more tracepoints just for the proper functioning of the signals endpoint. I do think we need some notion around what is loaded/supported at some point, as various eBPF programs will have different minimum kernel requirements, so we will likely expose a larger feature-set on more recent kernels.

I imagine at some point, we want to have remote eBPF deployment and instrumentation. Sort of an API for `bpftool` that tells you which programs are loaded, which maps are there etc. Maybe let's revisit health at that point and right now stick with "if it didn't error out, it loaded successfully"?